desc/derivatives/diffmat_utils.py

```diff
@@ -0,0 +1,306 @@
+with `jax.vmap` / `jax.pmap` inside larger optimisation loops.
+
+Highlights
+----------
+* **Chebyshev–Lobatto grid:**  ``chebpts_lobatto`` generates the collocation
+  nodes; ``diffmat1_lobatto`` and ``diffmat2_lobatto`` build the first‑ and
+  second‑derivative matrices in *O(n^2)* time using closed‑form formulas (no
+  explicit linear solves).
+* **Equispaced Fourier grid:**  ``fourier_pts`` returns the grid, while
+  ``fourier_diffmat*`` return the differentiation matrices.  Both even and odd
+  grid sizes are handled with the classic trigonometric explicit expressions.
+* **JIT compatibility:**  All functions work transparently under
+  ``jax.jit`` / ``jax.grad`` / ``jax.vmap`` because they avoid Python control
+  flow that depends on array values and use only JAX primitives.
+
+The implementations follow the formulas in
+  **Trefethen, L. N. (2000). *Spectral Methods in MATLAB*. SIAM** and
+  **Canuto et al. (2006). *Spectral Methods – Fundamentals in Single Domains*.**
+
+Examples
+--------
+>>> from desc.backend import jnp
+>>> from jax import jit
+>>> D1 = jit(diffmat1_lobatto)(16)   # first‑derivative, 16‑point grid
+>>> x  = chebpts_lobatto(16)
+>>> f  = jnp.sin(jnp.pi * x)
+>>> df = D1 @ f                     # approximate derivative at the nodes
+"""
+
+from desc.backend import jnp, vmap
+
+########################################################################
+# ---------------------- CHEBYSHEV MATRICES --------------------------- #
+########################################################################
+
+
+def chebpts_lobatto(n: int, domain=None):
+    """Return *n* Chebyshev–Lobatto points.
+
+    The nodes are given by
+        x_k = sin( (pi / 2) * (2 k - n + 1) / (n - 1) ),   k = 0, …, n‑1,
+    mapped linearly to the user‑supplied *domain*.
+
+    Parameters
+    ----------
+    n : int
+        Number of collocation points.  If *n* <= 0 an empty array is returned.
+    domain : sequence of 2 floats, optional
+        Physical interval ``[a, b]``.  Defaults to ``[-1, 1]``.
+
+    Returns
+    -------
+    jax.Array
+        1‑D array of shape ``(n,)`` containing the grid points, using the active
+        default floating‑point precision.
+
+    Notes
+    -----
+    * The mapping is performed with a single fused JAX expression, ensuring the
+      function is fully JIT‑traceable.
+    * Endpoint weights ``c`` required for the differentiation formulas are **not**
+      returned – they are (re)computed internally by the differentiation helpers.
+    """
+    if n <= 0:
+        return jnp.array([])
+
+    if domain is None:
+        domain = [-1, 1]
+
+    k = jnp.arange(n)
+    x = jnp.sin(jnp.pi * jnp.flip(2 * k - n + 1) / (2 * (n - 1)))
+
+    # Affine map to [a, b] only when necessary (allows JIT constant folding)
+    if domain[0] != -1 or domain[1] != 1:
+        x = (domain[1] - domain[0]) / 2 * (x + 1) + domain[0]
+    return x
+
+
+def diffmat1_lobatto(n: int):
```

Comment:
@ddudt show example of using our Transform classes to form these types of matrices